### PR TITLE
Dint 57

### PR DESCRIPTION
**Pull request by grahamcrackers**
## Description

Adding shoppingCartView event to aep collector

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [x] I have updated the documentation accordingly.
-   [x] I have read the **CONTRIBUTING** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.

### CODE REVIEW

File: src/types/aep/idAndValue.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// i really don't know what else to call this class, `Base` sounds way to generic
```
Comment (aniham) [inline src/types/aep/idAndValue.ts:1]: i don't see this class as a useful abstraction and think we should do it this way:

https://github.com/adobe/magento-storefront-event-collector/blob/main/src/types/aep/commerce.ts#L43-L45

File: src/handlers/shoppingCart/viewAEP.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        return undefined;
+    }
+
+    return cart.items.map(item => ({
+        SKU: item.product.sku,
+        name: item.product.name,
+        quantity: item.quantity,
+        priceTotal: item.prices.price.value,
+        currencyCode: item.prices.price.currency,
+    }));
```
Comment (aniham) [inline src/handlers/shoppingCart/viewAEP.ts:26]: we should use our utils method to get these:

https://github.com/adobe/magento-storefront-event-collector/blob/main/src/utils/aep/productListItems.ts

File: src/types/aep/commerce.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
+import { ProductListAdds, ProductListViews } from "./addToCart";
```
Comment (aniham) [inline src/types/aep/commerce.ts:1]: `ProductListAdds` is specific to `Commerce` - it's unnecessary and confusing to split it into a separate file

File: src/handlers/shoppingCart/viewAEP.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+/** handles shoppingCartView event that is sent to AEP */
+const aepHandler = async (event: Event): Promise<void> => {
+    const alloy = await getAlloy();
+
+    const { shoppingCartContext } = event.eventInfo;
+
+    // split this out later but i just need the id for now
+    const payload: BeaconSchema = {
+        commerce: {
```
Comment (aniham) [inline src/handlers/shoppingCart/viewAEP.ts:37]: i think we need a cartId + other cart info in the schema here, right?

**Review by aniham — CHANGES_REQUESTED** [review]
@grahamcrackers good stuff!! request a few changes, looks good otherwise!

File: src/handlers/shoppingCart/viewAEP.ts [hunk 5]
```diff
@@ -1,51 +1,34 @@
...
         commerce: {
             productListViews: {
-                // mark as 0 if id doesnt exist to easily scan in logs
-                id: shoppingCartContext.id?.toString() ?? "0",
+                id: "1",
                 value: 1,
             },
+            cart: {
+                cartId: shoppingCartContext.id ?? "1",
+            },
```
Comment (aniham) [inline src/handlers/shoppingCart/viewAEP.ts:25]: note to @grahamcrackers + @jekelly-adobe - i know this is wrong based on the discussion that happened 5 minutes ago, but i think it's ok to merge - i've already made the changes to my pr and waiting on a test to go through. when graham merges, i can pull down his stuff too and edit this one field (requires a schema edit too)

**Review by aniham — APPROVED** [review]

File: src/handlers/shoppingCart/viewAEP.ts (hunk 5 above)
Comment (grahamcrackers) [reply src/handlers/shoppingCart/viewAEP.ts:25]: 👍

**Pull request merged**